syntax = "proto3";

package events;

enum UserRole {
    OWNER = 0;
    EDITOR = 1;
    VIEWER = 2;
}

enum ObjectType {
    CARD = 0;
    STICKY_NOTE = 1;
    TEXT = 2;
    IMAGE = 3;
}

enum InputType {
    SIMPLE_SELECT = 0;
    MOVE = 1;
    ROTATE = 2;
    SCALE = 3;
}

message EventContext { // Didn't succeed to move it to separate file (import was not working)
    optional int64 sent_at = 1;
    optional int64 received_at = 2;
    optional int64 processed_at = 3;
    optional string message_id = 4;
    optional string user_agent = 5;
}

message TestEvent {
    optional EventContext context = 1;
    optional string event_name = 2; 
    // Data fields
    optional string user_id = 3;
    optional string account_id = 4;
    optional UserRole user_role = 5;
}

message YetAnotherTestEvent {
    optional EventContext context = 1;
    optional string event_name = 2; 
    // Data fields
    optional InputType input_type = 3;
    optional string object_id = 4;
    optional ObjectType object_type = 5;
}

